tempesta_fw/tls.c

```diff
@@ -412,9 +404,14 @@ tfw_tls_do_cleanup(void)
 static int
 tfw_tls_start(void)
 {
-	int r;
+	int r = tfw_runstate_is_reconfig();
 
-	if (tfw_runstate_is_reconfig())
+	if (!tfw_tls.crt.version) {
+		TFW_ERR("TLS: please spcify a certificate with"
+			" tls_certificate configuration option\n");
+		return -EINVAL;
```

Comment:
`tfw_tls.crt.version == NULL` is not an error if there is no listeners for https protocols. Simple `return 0` must be here. E.g. I have the following configuration:
```
server 127.0.0.1:8080;

vhost default {
    proxy_pass default;
}
```
It's absolutely valid: https protocol is not used, and no certificate defined.  But with your patch i cant start Tempesta:
```
[tempesta] Prepearing for the configuration processing.
[tempesta] Configuration processing is completed.
[tempesta] ERROR: TLS: please spcify a certificate with tls_certificate configuration option
[tempesta] ERROR: Unable to start module 'tls': -22
[tempesta] Stopping all modules...
[tempesta] modules are stopped
[tempesta] Warning: Configuration parsing has failed. Clean up...
```

I've added `cfgend()` function into `tls.c` not so many time ago. It checks for misconfiguration errors after configuration file is fully parsed. It's better to check during `cfgend()` that the certificate is configured when https protocol is to be used.

I also think that configuration errors must be checked ant `cfgend` stage and processing them on `start` should be discouraged. There is no sense in starting everything with invalid configuration.